### ISSUE TITLE
Citrix Receiver client turns black when client on another monitor is focused.

### ISSUE THREAD

**Issue by inspell** [awesomeWM/awesome#2133]
Output of `awesome --version`:

awesome v4.0 (Harder, Better, Faster, Stronger)
 • Compiled against Lua 5.1.5 (running with Lua 5.1)
 • D-Bus support: ✔
 • execinfo support: ✔
 • RandR 1.5 support: ✔
 • LGI version: 0.9.1

**How to reproduce the issue:**

Open Citrix Receiver and position it on Monitor 1.
Open a client (terminal for example) on Monitor 2 and focus it.

**Actual result:**

Citrix Receiver client turns black.

**Expected result:**

Citrix Receiver should still be visible.

Here are some more details:
I am running Debian 9.1 stretch x64 with kernel:
4.9.0-3-amd64 SMP Debian 4.9.30-2+deb9u5 (2017-09-19) x86_64 GNU/Linux
I am using two-monitor setup.

I am using Citrix Receiver to log in to a remote system.
I use the receiver in a Windowed mode (not fullscreen).
The first odd thing is that the Citrix Receiver window (client) can only be placed on Monitor 1.
When I move it to Monitor 2 it instantly moves itself back to Monitor 1.
I cannot even change its place inside Monitor 1 too. The moment I try to change its position it automatically goes to the left top corner.
And the second thing, which is most annoying, is that when I focus a client on Monitor 2, the client of Citrix Receiver on Monitor 1 turns black and nothing is shown. This doesn't happen if I focus another client (not Citrix Receiver) inside Monitor 1.

Any suggestions on how to debug that?
Can I attach some hook to the Citrix Receiver client which will repaint the client when it loses focus or something similar?

![selected-client-on-monitor1](https://user-images.githubusercontent.com/3369373/33552789-e853b750-d8fe-11e7-9762-abc3aa801aaf.png)
![selected-client-on-monitor2](https://user-images.githubusercontent.com/3369373/33552793-ee18473c-d8fe-11e7-8941-29dc45d4f8ae.png)

**Comment by psychon** [comment]
> Any suggestions on how to debug that?

The only way that a window can (realistically) turn black is because it paints itself black. Why it would do that, I don't know, but by the rest of your descriptions, the authors of this software seem to have some "special ideas"...
So, to answer your question: I fear it's not a bug, but a future.

How's the behavior in other WMs? Does the same happen there, too?

Also, I briefly tried to figure out what Citrix Receiver is and what it's license is, but basically failed to find some answers (well, I feel like the answer is "it's complicated").

**Comment by inspell** [comment]
Yes, it is complicated ; )
I don't even know how to configure the server side, I just use it in the office. It basically allows you to log in remotely to a machine. I will try to see how it behaves with Gnome and let you know.
At first I thought it is a security feature, but this didn't happen on the old Debian that I had but it also had an older Awesome WM version and also probably older Citrix Receiver version.

**Comment by inspell** [comment]
I tried it on Gnome and the problem didn't appear. The client didn't turn black.
I was also able to move it from monitor 1 to monitor 2 and back and the client didn't automatically go to the top-left corner of the screen. When I am using the Citrix Receiver with Awesome WM I am using the floating layout (not tiling).

So, there must be something that happens when I am using Awesome WM that turns my Citrix Receiver client black. Suggestions and hints are welcome.

**Comment by b01t** [comment]
Does black happen to be the desktop background color on the remote machine? Not the background image, if any, but the underlying background color? The Citrix client I've used for Windows seems to attempt to avoid remoting and goes completely blue (desktop background color) when I remote into the machine running the client by using either VNC or TeamViewer.

Just a thought.

**Comment by inspell** [comment]
Yes, I checked it just now and the Solid Background Color is Black. Unfortunately I cannot change it to something else to verify that it will reflect the client's color - it seems that I am less than a regular user on this machine and I cannot change almost anything. This is interesting but it doesn't explain why Gnome doesn't have this problem.
